### PR TITLE
add finnish paraphrase-based STS dataset

### PR DESCRIPTION
**Pull request by rbroc**
## Checklist for adding MMTEB dataset

Reason for dataset addition: First Finnish STS dataset

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by rbroc** [comment]
A note on this: performance scores are much lower than other STS tasks. I have run an additional test though, testing a monolingual finnish model, https://huggingface.co/TurkuNLP/sbert-uncased-finnish-paraphrase, which gets similarity scores between .3 - .4. 

Furthermore, the similarity scores obtained with multilingual encoders should be statistically significant (p ~ 0.001), if that's a good definition of better than chance.

In my view, just a challenging dataset (most labels are 3 and 4, and the different between the two is `3 = good paraphrase in this context`, but not in general; `4 = good paraphrase in all contexts`), just calling for a very nuance semantic understanding.

But happy to hear what y'all think.

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (isaac-chung) [inline mteb/tasks/STS/fin/FinParaSTS.py:57]: nit: use a variable `n_samples=1000`

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (isaac-chung) [inline mteb/tasks/STS/fin/FinParaSTS.py:71]: same here for 1000.

**Review by isaac-chung — APPROVED** [review]
Looks good to me, but I would like a second opinion from one of the maintainers: @KennethEnevoldsen or @imenelydiaker 
 maybe. 
I'm unsure (for my own curiosity) about the [extra label 4 flags](https://aclanthology.org/2021.nodalida-main.29.pdf)' role in the scoring:
```
- Subsumption (> or <) w
- Style (s)
- Minor deviation (i)
```

**Comment by rbroc** [comment]
thanks a lot @isaac-chung! Re: the extra labels, I have simply stripped additional annotations, and left the raw scores as labels, but curious to hear what people think. :)

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (imenelydiaker) [inline mteb/tasks/STS/fin/FinParaSTS.py:75]: Very nice! You're handling the string `label` conversion to integers 🙂

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (rbroc) [reply mteb/tasks/STS/fin/FinParaSTS.py:57]: pydantic throws an error, as this must be a dictionary. will keep it as is :)

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (rbroc) [reply mteb/tasks/STS/fin/FinParaSTS.py:71]: resolving as per previous comment

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (rbroc) [reply mteb/tasks/STS/fin/FinParaSTS.py:75]: @imenelydiaker self.dataset is a `DatasetDict`, and `select` cannot be applied to that -- we could do this as part of map, but it would be cumbersome and subsetting early on increases efficiency. 
would strongly lean towards leaving as is.

**Comment by rbroc** [comment]
> Is this dataset a subset/duplication of [GEM/opusparcus](https://huggingface.co/datasets/GEM/opusparcus)? In which case, Opusparcus is already handled in mteb and handles Finnish. Can you please check? Because I see a `gem_id`, which refers to the id of the item in the GEM dataset, maybe we can filter samples that are already handled by Opusparcus using the `gem_id` column ?

I have double-checked and it is not. `GEM` refers to the data format for OpenSubtitles, which both use to get `sentence1` (though very different subsets). Paraphrases are then generate with automated translation of parallel examples for opusparcus, while here they use alternative versions of Finnish subtitles. Also, this dataset includes news on top of subtitles.

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (rbroc) [reply mteb/tasks/STS/fin/FinParaSTS.py]: loading `plain` now (i was loading `classification` before). it is done by passing `name` to the `dataset` dict in the metadata, as `dataset` is passed as kwargs to `load_data` (see https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/AbsTask.py#L35), and `name` specifies the config to pull. this should be ready to be resolved.

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (rbroc) [reply mteb/tasks/STS/fin/FinParaSTS.py:75]: we're pulling main now, hence ready to resolve

**Comment by rbroc** [comment]
@imenelydiaker @isaac-chung I've addressed your points and resolved some of the things that can't be addressed (e.g., can't pass `n_samples=1000`, `pydantic` requires a dict). I've rerun evaluations too and added points.

File: mteb/tasks/STS/fin/FinParaSTS.py
Comment (isaac-chung) [reply mteb/tasks/STS/fin/FinParaSTS.py:57]: Oh sorry. I meant something like this, and defining `N_SAMPLES=1000` around line 6 or 7.
```suggestion
        n_samples={"test": N_SAMPLES, "validation": N_SAMPLES},
```

**Comment by isaac-chung** [comment]
Thanks! I clarified on my comments, but this still looks good to me. If it's all good from @imenelydiaker's side too then we can merge :)

**Review by imenelydiaker — APPROVED** [review]
Thanks for the clear answers, great job! We can merge 🙂

**Comment by rbroc** [comment]
thanks @imenelydiaker and @isaac-chung for a very thorough review 🚀 I have made one more tiny change, passing `n_samples` as a variable as suggested by @isaac-chung (sorry for the misunderstanding! :)) and updating avg length after having decided to pull the `plain` subset. 

should be ready to merge as far as I am concerned too :)

**Comment by rbroc** [comment]
@imenelydiaker @isaac-chung now tests on 3.8 are failing, but the errors do not seem to be dependent on anything related to this PR. it seems like it's skipping one the workflows though as a consequence of this one failing.

looks like it's something funky with caching though, can one of your trigger the workflows again to see if it's just an accident?

**Comment by imenelydiaker** [comment]
I think it's related to this issue: https://github.com/embeddings-benchmark/mteb/issues/454. I see in the logs that it's related to markdown, maybe we can bypass it?

cc @KennethEnevoldsen

**Comment by imenelydiaker** [comment]
I restarted the windows job with debug meanwhile

**Comment by rbroc** [comment]
awesome! everything's passing now :)

**Comment by imenelydiaker** [comment]
I'm merging, thank you again for your contribution @rbroc !

**Pull request merged**
